sippy-ng/src/component_readiness/IncludeVariantCheckboxList.js

```diff
@@ -1,31 +1,190 @@
+      ? varsContext.includeVariantsCheckedItems[variantGroupName]
       : []
   )
+  const updateIncludeItems = (newItems) => {
+    varsContext.replaceIncludeVariantsCheckedItems(variantGroupName, newItems)
+    setIncludeItems(newItems)
+  }
+
+  const [compareItems, setCompareItems] = React.useState(
+    variantGroupName in varsContext.compareVariantsCheckedItems
+      ? varsContext.compareVariantsCheckedItems[variantGroupName]
+      : []
+  )
+  const updateCompareItems = (newItems) => {
+    varsContext.replaceCompareVariantsCheckedItems(variantGroupName, newItems)
+    setCompareItems(newItems)
+  }
+
+  const handleVariantSelection = (
+    event,
+    checkedItems,
+    setCheckedItems,
+    updateCheckedItems
+  ) => {
+    const item = event.target.name
+    const newItems = event.target.checked
+      ? [...checkedItems, item]
+      : checkedItems.filter((checkedItem) => checkedItem !== item)
+    setCheckedItems(newItems)
+    updateCheckedItems(newItems)
+  }
+  const handleIncludeVariantSelection = (event) => {
+    handleVariantSelection(
+      event,
+      includeItems,
+      setIncludeItems,
+      updateIncludeItems
+    )
+  }
+  const handleCompareVariantSelection = (event) => {
+    handleVariantSelection(
+      event,
+      compareItems,
+      setCompareItems,
+      updateCompareItems
+    )
+  }
+
+  const [isCompareMode, setIsCompareMode] = useState(
+    varsContext.variantCrossCompare.includes(variantGroupName)
+  )
+  const handleToggleCompare = (event) => {
+    varsContext.updateVariantCrossCompare(variantGroupName, !isCompareMode)
+    setIsCompareMode(!isCompareMode)
+  }
+
+  const classes = makeStyles((theme) => ({
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: '20px',
+    },
+    gridCenter: {
+      marginTop: theme.spacing(1),
+      textAlign: 'center',
+    },
+    gridRight: {
+      justifyContent: 'flex-end',
+      display: 'flex',
+    },
+  }))()
 
-  const updateCheckedItems = (newCheckedItems) => {
-    varsContext.replaceIncludeVariantsCheckedItems(variantName, newCheckedItems)
-    setCheckedItems(newCheckedItems)
+  let params = {
+    variantGroupName,
+    classes,
+    displayList: varsContext.allJobVariants[variantGroupName],
+    includeItems,
+    handleIncludeVariantSelection,
+    compareItems,
+    handleCompareVariantSelection,
   }
+
+  return (
+    <FormControl
+      variant="standard"
+      className={classes.formControl}
+      component="fieldset"
+    >
+      <Accordion className="checkboxlist-headerName">
+        <AccordionSummary
+          className="checkboxlist-summary"
+          expandIcon={<ExpandMore />}
+        >
+          <Typography className="checkboxlist-label">
+            Include {variantGroupName}
+          </Typography>
+        </AccordionSummary>
+        <AccordionDetails>
+          {isCompareMode
+            ? CompareVariantGroup(params)
+            : IncludeVariantGroup(params)}
```

Comment:
TIL, this is cool.

Reply:
what, the ternary?

mainly just trying to make it readable... very messy to mix logic and elements

Reply:
No the functions injecting more jsx, didn't realize we could do that.